metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentData.groovy

```diff
@@ -5,6 +5,13 @@ import life.qbic.metastats.datamodel.MetaStatsSample
 
 interface FilterExperimentData {
 
-    def filterProjectMetaData(List<MetaStatsSample> samples, List<MetaStatsExperiment> experiments)
+    /**
+     * Method to filter information required for the MetaStats output from samples and experiments and transfering it into
+     * a MetaStatsPackageEntry object
+     *
+     * @param testSamples describing Q_TEST_SAMPLES of the project with all information of related samples
+     * @param experiments that are part of the project
+     */
+    void filterProjectMetaData(List<MetaStatsSample> testSamples, List<MetaStatsExperiment> experiments)
```

Comment:
`filter`XYZ`(Closure criteria)`

When I see a filter method I would like to be able to specify a filter to apply. In this case the Filter interface might be implemented in a configurable filter or the filter implementing class has some configuration. I would rename it to `Filter filter = new Filter(); filter.apply() ` or `filter.filterProjectMetadata(ProjectMetadata projectMetadata)`

Reply:
67d3316

Reply:
nice solution